ecommerce/core/templatetags/core_extras.py

```diff
@@ -31,6 +36,90 @@ def settings_value(name):
     return getattr(settings, name)
 
 
+@register.simple_tag(name='is_special_coupon')
+def is_special_coupon(object_passed):
+    """
+    Checks if the coupon name meets the format criteria of a special coupon. The coupon name is
+    extracted from the 'object_passed' argument which could be an Order, Basket or OrderDiscount instance.
+
+    Arguments:
+        object_passed (Order or SimpleLazyObject or OrderDiscount): Order of the purchase, its basket or its OrderDiscount.
+
+    Returns:
+        bool: True if the coupon name is a valid special coupon, False otherwise.
+    """
+    is_order = isinstance(object_passed, Order)
+    is_basket = isinstance(object_passed, SimpleLazyObject)
+    is_order_discount = isinstance(object_passed, OrderDiscount)
+    coupon_name = ''
+
+    if not is_order and not is_basket and not is_order_discount:
+        return False
+
+    if is_order:
+        discount = object_passed.basket_discounts.first()
+        coupon_name = discount.voucher.name if discount else ''
+
+    if is_basket:
+        coupon = object_passed.vouchers.first()
+        coupon_name = coupon.name if coupon else ''
+
+    if is_order_discount:
+        coupon_name = object_passed.voucher.name
+
+    if not coupon_name:
+        return False
+
+    request = get_current_request()
```

Comment:
You need to check that a site/siteconfiguration has been set for request. @Jacatove 
This variable is only used once.

Reply:
Add a blank line after this line. @Jacatove